### ISSUE TITLE
No EntityDamageEvent called for EnderCrystal if the DamageSource isn't entity related

### ISSUE THREAD

**Issue by Trigary** [PaperMC/Paper#4193]
### What behaviour is expected:
EntityDamageEvent being called for EnderCrystal, even if the DamageSource is not an instance of EntityDamageSource

### What behaviour is observed:
EntityDamageEvent isn't called for EnderCrystal, because of this line of code: https://hub.spigotmc.org/stash/projects/SPIGOT/repos/craftbukkit/browse/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java#989

### Steps/models to reproduce:
 - Place and ender crystal and "install" a plugin that prints out all EnderCrystal related EntityDamageEvents.
 - Placing a TNT next to the crystal and igniting it using flint and steel causes the EntityDamageEvent to be called.
 - Placing a TNT next to the crystal and igniting it using a fire arrow from a dispenser causes the end crystal to be destroyed without the event being called.
 - Placing another end crystal next to the crystal and hitting it causes the crystal that's not the one that was hit to be destroyed without the event being called.

### Plugin list:
Just the debugging plugin.

### Paper build number:
git-Paper-136 (MC: 1.16.1) (Implementing API version 1.16.1-R0.1-SNAPSHOT)

### Anything else:
Arrows from dispensers destroy end crystals in Vanilla, but not on Paper (they just pass through). Possible related, but I didn't look into it.

**Comment by Trigary** [comment]
Previously there were more checks in the linked code, not just `EntityEnderCrystal`. For some reason, they got removed. Would the solution be to remove this early return completely? Would that break anything?

Luckily this code is pretty contained:
 - only the ender crystal entity is affected
 - removing the check would only make more events fire, which has a high chance of not breaking anything
 - since non EntityDamageSource DamageSources can already traverse the codebase, similar non-EntityDamageSources-instances with the only difference in the entity (victim) shouldn't break anything (I searched for the usages of EntityEnderCrystal)

I shall sleep on it and probably test and then PR this simple change.
